src/NServiceBus.RavenDB/Outbox/OutboxRecordsCleaner.cs

```diff
@@ -15,61 +13,26 @@ public OutboxRecordsCleaner(IDocumentStore documentStore)
             this.documentStore = documentStore;
         }
 
-        public async Task RemoveEntriesOlderThan(DateTime dateTime)
+        public async Task RemoveEntriesOlderThan(DateTime dateTime, CancellationToken cancellationToken = default(CancellationToken))
         {
-            lock (this)
+            var query = new IndexQuery
             {
-                if (doingCleanup)
-                {
-                    return;
-                }
+                Query = $"DispatchedAt:[* TO {dateTime:o}]"
+            };
 
-                doingCleanup = true;
-            }
-
-            try
+            var bulkOpts = new BulkOperationOptions
             {
-                var deletionCommands = new List<ICommandData>();
-
-                using (var session = documentStore.OpenAsyncSession())
-                {
-                    var query = session.Query<OutboxRecord, OutboxRecordsIndex>()
-                        .Where(o => o.Dispatched)
-                        .OrderBy(o => o.DispatchedAt);
-
-                    using (var enumerator = await session.Advanced.StreamAsync(query).ConfigureAwait(false))
-                    {
-                        while (await enumerator.MoveNextAsync().ConfigureAwait(false))
-                        {
-                            if (enumerator.Current.Document.DispatchedAt >= dateTime)
-                            {
-                                break; // break streaming if we went past the threshold
-                            }
+                AllowStale = true
+            };
 
-                            /*
-                             * For some unknown reasons when streaming classes with no
-                             * Id property the Key is not filled. The document Id can be
-                             * found in the streamed document metadata.
-                             */
-                            var key = enumerator.Current.Key ?? enumerator.Current.Metadata["@id"].ToString();
+            var operation = await documentStore.AsyncDatabaseCommands.DeleteByIndexAsync(nameof(OutboxRecordsIndex), query, bulkOpts, cancellationToken)
```

Comment:
`nameof` here might be dangerous since we are starting from the assumption that we won't change/customize the index name in any way. E.g. if the call will be named `Outbox_RecordsIndex` the index name will be `Outbox/RecordsIndex` in RavenDB.